src/main/java/sirius/biz/web/BasePageHelper.java

```diff
@@ -411,11 +412,12 @@ protected void enforcePaging(Page<E> result, List<E> items) {
         if (originalSize > pageSize) {
             result.withHasMore(true);
             items.remove(items.size() - 1);
-        }
-        if (fetchTotalCount && (originalSize > pageSize || result.getStart() > pageSize)) {
-            result.withTotalItems((int) baseQuery.count());
-        } else {
-            result.withTotalItems(items.size());
+            if (withTotalCount) {
+                result.withTotalItems((int) baseQuery.count());
+            }
+        } else if (withTotalCount) {
+            // we don't have any more items, so total items is end of the current page
+            result.withTotalItems(result.getEnd());
```

Comment:
isn't here an off-by-1 if originalSize == pageSize (?)  - i think thats the whole point of keeping originalSize around

Reply:
Not really if originalSize == pageSize we never remove anything from items so it doesnt matter if we use originalSize or items.size()

But i found at this point the items aren't added into the result so we can't use result.getEnd() anyway and spell it out